extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -483,6 +493,95 @@ public static KeyOrValueSpec avroSpec(final String schemaName) {
             return new AvroConsume(schemaName, AVRO_LATEST_VERSION, DIRECT_MAPPING);
         }
 
+        /**
+         * Avro spec from an Avro schema. String fields will be not be converted to Java Strings.
+         *
+         * @param schema An Avro schema.
+         * @param fieldNameToColumnName A mapping specifying which Avro fields to include and what column name to use
+         *        for them; fields mapped to null are excluded.
+         * @return A spec corresponding to the schema provided.
+         */
+        @SuppressWarnings("unused")
+        public static KeyOrValueSpec avroSpecWithUTF8Strings(final Schema schema,
+                final Function<String, String> fieldNameToColumnName) {
+            return new AvroConsume(schema, fieldNameToColumnName, true);
+        }
+
+        /**
+         * Avro spec from an Avro schema. All fields in the schema are mapped to columns of the same name. String fields
+         * will be not be converted to Java Strings.
+         *
+         * @param schema An Avro schema.
+         * @return A spec corresponding to the schema provided.
+         */
+        @SuppressWarnings("unused")
+        public static KeyOrValueSpec avroSpecWithUTF8Strings(final Schema schema) {
+            return new AvroConsume(schema, DIRECT_MAPPING, true);
+        }
+
+        /**
+         * Avro spec from fetching an Avro schema from a Confluent compatible Schema Server. The Properties used to
+         * initialize Kafka should contain the URL for the Schema Server to use under the "schema.registry.url"
+         * property. String fields will be not be converted to Java Strings.
+         *
+         * @param schemaName The registered name for the schema on Schema Server
+         * @param schemaVersion The version to fetch
+         * @param fieldNameToColumnName A mapping specifying which Avro fields to include and what column name to use
+         *        for them; fields mapped to null are excluded.
+         * @return A spec corresponding to the schema provided.
+         */
+        @SuppressWarnings("unused")
+        public static KeyOrValueSpec avroSpecWithUTF8Strings(final String schemaName,
+                final String schemaVersion,
+                final Function<String, String> fieldNameToColumnName) {
+            return new AvroConsume(schemaName, schemaVersion, fieldNameToColumnName, true);
+        }
```

Comment:
Not a big fan of the proliferation of `WithUTF8Strings` messages; arguably, we may want to follow more of a builder pattern for cases like this in the future.

What about creating only having a single new method `avroSpec(String, String, Function, boolean)` that mirrors the new AvroConsume constructor?

Reply:
Sure. Should be sufficient.